blocks/algorithms/__init__.py

```diff
@@ -222,6 +224,29 @@ def compute_step(self, param, grad_wr_param):
         """
         raise NotImplementedError
 
+    def compute_steps(self, gradients):
+        """Build a Theano expression for steps for all parameters.
+
+        Override this method if you want to process the gradient
+        with respect to all parameters as a whole, not parameter-wise.
+
+        Parameters
+        ----------
+        gradients : dict of (:class:`~tensor.TensorSharedVariable`
```

Comment:
And this is exactly what I call an unnecessary complication. Just compare the two constructions:

```python
{param: self.compute_step(param, gradients[param]) for param in gradients}
```

```python
OrderedDict((param, self.compute_step(param, gradients[param]) for param in gradients)
```

The first one uses Python nice built-in syntax for dictionaries and is more readable.